test/ussd_optout_rapidpro.test.js

```diff
@@ -8,25 +10,748 @@ describe("ussd_optout_rapidpro app", function() {
+        });
+        it("should go to state_no_previous_optout if user hase no subscriptions and not " +
+            "previously opted out", function() {
+            return tester
+                .setup(function(api) {
+                    api.http.fixtures.add(
+                        fixtures_rapidpro.get_contact({
+                            urn: "tel:+27123456789",
+                            exists: true,
+                            language: "zul",
+                            groups: []
+                        })
+                    );
+                })
+                .start()
+                .check.interaction({
+                    state: "state_no_previous_optout",
+                    reply: [
+                        "Welcome to the Department of Health's MomConnect. " +
+                        "Dial *154*550*2# when you are at a clinic to sign up to " +
+                        "receive helpful messages for you and your baby."
+                    ].join("\n")
+                })
+                .check.user.lang("zul")
+                .run();
+        });
+    });
+    describe("state_opt_out", function() {
+        it("should display the options to the user", function() {
+            return tester
+                .setup.user.state("state_opt_out")
+                .input({session_event: "continue"})
+                .check.interaction({
+                    state: "state_opt_out",
+                    reply: [
+                        "Hello mom! Do you want to stop getting MomConnect (MC) messages?",
+                        "1. Yes",
+                        "2. No"
+                    ].join("\n")
+                })
+                .run();
+        });
+        it("should display the error pretext if the user types an invalid choice", function() {
+            return tester
+                .setup.user.state("state_opt_out")
+                .input("foo")
+                .check.interaction({
+                    state: "state_opt_out",
+                    reply: [
+                        "Sorry, please reply with the number next to your answer. " +
+                        "Do you want to stop getting MomConnect (MC) messages?",
+                        "1. Yes",
+                        "2. No"
+                    ].join("\n")
+                })
+                .run();
+        });
+        it("should go to state_no_optout if they choose no", function() {
+            return tester
+                .setup.user.state("state_opt_out")
+                .input("2")
+                .check.interaction({
+                    state: "state_no_optout",
+                    reply: 
+                        "Thanks! MomConnect will continue to send you helpful messages and process your " +
+                        "personal info. Have a lovely day!"
+                })
+                .run();
+        });
+        it("should go to state_delete_research_info if they choose yes", function() {
```

Comment:
should go to state_optout_reason if they choose yes

Reply:
ahh, thanks